### ISSUE TITLE
Add search

### ISSUE THREAD

**Issue by bluepichu** [elmzteam/bydesign#19]
As discussed in #18.

**Comment by zwade** [comment]
So right now, posts are stored in the fs for reliability reasons. However, since mongo can't index them, it makes it a pain in the neck to do searching. The way I see it, we have three options
- Keep everything as it is, but not have search functionality over post content
- Move post content to the db for easy searching, but risk volatility
- On startup, duplicate fs into db for indexing, which is somewhat wasteful, and makes content management a pain

Thoughts?

**Comment by el1t** [comment]
> Stores data in database
> 
> More volatile than fs

Shouldn't it be the other way around?

**Comment by bluepichu** [comment]
@zwade we should revisit this, particularly since you're already doing a bunch of post restructuring anyway.

**Comment by zwade** [comment]
Yeah, it might not be a terrible idea to move posts into the database. I still kinda prefer the idea of a hard copy (in a cash-under-the-mattress sort of way), but I think the benefits of a transition outway the potential risk

**Comment by zwade** [comment]
With that in mind though, it would be easier to do that before dealing with slugs and stuff, so I may push the guid version first, and have slugs in a different PR.

**Comment by bluepichu** [comment]
This should be relatively straightforward using mongo's built-in text score once #106 goes through.  However, we'll probably also want to add in extra relevance weight for tags, title, and author name.
